urllib3/connectionpool.py

```diff
@@ -396,7 +396,7 @@ def _make_request(self, conn, method, url, timeout=_Default, chunked=False,
 
         try:
             assert_header_parsing(httplib_response.msg)
-        except HeaderParsingError as hpe:  # Platform-specific: Python 3
+        except (HeaderParsingError, TypeError) as hpe:  # Platform-specific: Python 3
```

Comment:
Dear reviewer: This was questionable. The code patch for `assert_header_parsing` is only supposed to matter on Python 3, however, it was borking my tests due to a TypeError. If you can think of a better way to fix this, I'm all ears.